### PR TITLE
 Migrate tide to use the new k8s clients 

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov skuznets@redhat.com

Depends on #11060

/assign @cjwagner @fejta @munnerz
/cc @sebastienvas @BenTheElder @krzyzacy

### CODE REVIEW

File: prow/tide/tide_test.go [hunk 1]
```diff
@@ -1042,24 +1031,32 @@ func TestTakeAction(t *testing.T) {
...
-		if tc.triggered != len(fkc.createdJobs) {
-			t.Errorf("Wrong number of jobs triggered. Got %d, expected %d.", len(fkc.createdJobs), tc.triggered)
+
+		numCreated := 0
+		var batchJobs []*prowapi.ProwJob
+		for _, action := range fakeProwJobClient.Actions() {
+			switch action := action.(type) {
+			case clienttesting.CreateActionImpl:
+				numCreated++
+				if prowJob, ok := action.Object.(*prowapi.ProwJob); ok && prowJob.Spec.Type == prowapi.BatchJob {
```
Comment (fejta) [inline prow/tide/tide_test.go]: would also switch this to if !ok || pJ.S.T != pa.BJ { continue }

File: prow/tide/tide_test.go [hunk 2]
```diff
@@ -1042,24 +1031,32 @@ func TestTakeAction(t *testing.T) {
...
+					batchJobs = append(batchJobs, prowJob)
+				}
+			}
+		}
+		if tc.triggered != numCreated {
```
Comment (fejta) [inline prow/tide/tide_test.go]: What the difference between numCreated and batches? Why are we reordering all this?

File: prow/tide/tide_test.go (hunk 1 above)
Comment (stevekuznetsov) [reply prow/tide/tide_test.go]: we only do something in the positive case so that seems not most appropriate here

Comment (fejta) [reply prow/tide/tide_test.go]: Right, golang has a preference for identifying the negative case and exiting (and minimizing indentation of the positive code path).

Comment (stevekuznetsov) [reply prow/tide/tide_test.go]: right but we would have one line indented in both cases

Comment (stevekuznetsov) [reply prow/tide/tide_test.go]: and the `continue` case would be more LOC too

**Comment by stevekuznetsov** [comment]
@BenTheElder @krzyzacy @munnerz so @cjwagner is out for a week -- do y'all feel comfortable reviewing this or should we wait on him?

**Comment by krzyzacy** [comment]
(next prow bump will be exciting...)

**Pull request merged**